### ISSUE TITLE
No resource schema found for azurerm_capacity_reservation_group

### ISSUE THREAD

**Issue by mpjtaylor** [hashicorp/terraform-provider-azurerm#17455]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.7

### AzureRM Provider Version

3.11.0

### Affected Resource(s)/Data Source(s)

azurerm_capacity_reservation_group

### Terraform Configuration Files

```hcl
resource "azurerm_capacity_reservation_group" "rehosted_capacity_group" {
  name                = format("z%s%s%s%s%scrg%s", local.set2, local.set3, local.set4, local.set5, local.location_code, "90")
  resource_group_name = data.azurerm_resource_group.support.name
  location            = data.azurerm_resource_group.support.location
  zones               = ["1", "2", "3"]
  tags                = local.common_tags
}
resource "azurerm_capacity_reservation" "rehosted_capacity" {
  count                         = length(var.capacity_reservations)
  name                          = format("%s_%s", var.capacity_reservations[count.index].vmSize, var.capacity_reservations[count.index].zone)
  capacity_reservation_group_id = azurerm_capacity_reservation_group.rehosted_capacity_group.id
  zone                          = var.capacity_reservations[count.index].zone
  sku {
    capacity = var.capacity_reservations[count.index].quantity
    name     = var.capacity_reservations[count.index].vmSize
  }
  tags = local.common_tags
}
```

### Debug Output/Panic Output

```shell
Error: Missing resource schema from provider

No resource schema found for azapi_resource.
```

### Expected Behaviour

capacity group should have been removed

### Actual Behaviour

Error: Missing resource schema from provider

No resource schema found for azapi_resource.

### Steps to Reproduce

terraform apply
terraform destroy

**Comment by ms-henglu** [comment]
Hi @mpjtaylor ,

Thank you for opening this issue, from the error message, it seems that you're using `azapi_resource` but didn't specify the provider configuration?

**Comment by mpjtaylor** [comment]
Hi no not using azapi_resource using azurerm_capacity_reservation_group

**Comment by ms-henglu** [comment]
@mpjtaylor - Please check other tf files under the same directory, I guess there must be a `azapi_resource` which doesn't related to the `azurerm` provider.

**Comment by mpjtaylor** [comment]
yes defiantly using azurerm and azapi_resource but this error states azurerm_capacity_reservation_group
